planner/core/logical_plan_builder.go

```diff
@@ -3124,6 +3132,10 @@ func unfoldWildStar(field *ast.SelectField, outputName types.NameSlice, column [
 				}}
 			colName.SetType(col.GetType())
 			field := &ast.SelectField{Expr: colName}
+			// Alias name is only to ensure the view definition is correct.
+			if capFlag&canExpandAST != 0 && name.SelectField != nil {
+				name.SelectField.AsName = colName.Name.Name
```

Comment:
We are passing the `SelectField` as a pointer. Is this in-place modification safe?

Reply:
Here I want to change the `ast.SelectField` by adding an alias name. e.g. from `count(1)` to `count(1) AS count(*)`. So We have to use pointer make it take effects.

I don't very understand the specific meaning of safety.(like thread safety or some other scenes?)